views/search_view/views.py

```diff
@@ -0,0 +1,81 @@
+from django.shortcuts import render
+from django.views import View
+
+from core.user_controller.UserController import UserController
+from core.course_controller.CourseController import CourseController
+from ta_scheduler.models import Semester
+
+
+class SearchView(View):
+    def get(self, request, type: str):
+        """
+        GET request is used to initialize the page:
+        - If type is "course", all courses are displayed.
```

Comment:
can we make this in the precondition/postcondition format they want for the design doc

Reply:
same for other methods. I like your format better but it's what Rock wants :(